soc/nordic/nrf54h/power.h

```diff
@@ -17,4 +17,9 @@
  */
 void nrf_poweroff(void);
 
+/**
+ * @brief Power up and enable instruction and data cache.
+ */
+void nrf_cache_power_up(void);
```

Comment:
This name may be a bit confusing, as the `nrf_cache_` prefix is used by `hal/nrf_cache.h` in nrfx. Maybe `nrf_power_up_cache` could work better?